model/flow/mapfunc/identity.go

```diff
@@ -0,0 +1,12 @@
+package mapfunc
+
+import (
+	"github.com/onflow/flow-go/model/flow"
+)
+
+func WithStake(stake uint64) flow.IdentityMapFunc {
+	return func(identity flow.Identity) flow.Identity {
+		identity.Stake = stake
```

Comment:
`IdentityMapFunc` takes a `flow.Identity` rather than a `*flow.Identity`, so the structure is copied into the mapping function (in `Map` we do `f(*identity)`). So for literal fields this will work the way we want and not modify the original data. 

We will still have the bug you're mentioning if we modify fields containing pointers, so it is better to use a deep copy like you mentioned. Practically the two fields that could cause this are the two public key fields. These types don't have a copy method currently and embed CGO types which would need to be copied as well. 

I will at least make the `Map` function documentation more clear on its behaviour w.r.t. modifying the input. cc @tarakby for input on copying public keys from the crypto API side

Reply:
https://github.com/onflow/flow-go/pull/143/commits/0a4c1a8f4fe871e6df669fb77f92da50b63dc692

Reply:
Thanks for adding the comment highlighting the go's copying on pass by value. Helps people like me who are less proficient with go.